### PR TITLE
Movement

### PR DESCRIPTION
**Pull request by gdxn96**

### CODE REVIEW

File: Spikes/movement/Character.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	m_target.setFillColor(yellow);
+	m_target.setPosition(Vector2f(m_targetPos.x - m_targetRadius, 
+								  m_targetPos.y - m_targetRadius));
+}
+
+
+void Character::update()
+{
+	resetAngles();	//if above 360 or below 0 degrees
+
```
Comment (gdxn96) [inline Spikes/movement/Character.cpp]: does reset angles need to be called in update or only whenever you activate the pointer?

File: Spikes/movement/Character.cpp [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	{
+		m_angle = m_targetAngle;
+		m_move = false;
+	}
+
+
+	if (m_move)
+	{
+		m_angle += m_speed;
+		m_pos = Vector2f(m_centre.x + PATH_RADIUS * cos(m_angle),
```
Comment (gdxn96) [inline Spikes/movement/Character.cpp]: probably should either comment this or break it up into smaller chunks i.e. xOffset = PathRadius \* cos(m_angle); use your best gudgement (geay key not working)

File: Spikes/movement/Character.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	Vector2f m_pos;
+	float m_radius;
+	float m_speed;
+	float m_angle;
+	CircleShape m_shape;
+
+	//Target
+	Vector2f m_targetPos; 
+	float m_targetRadius;
+	float m_targetAngle;
```
Comment (gdxn96) [inline Spikes/movement/Character.h]: What is target radius? Make it apparent, also I would think that the target shape you are creating/drawing could be actually implemented elsewhere as opposed to the player/character.cpp. Instead character should maybe have a target pos instead.

File: Spikes/movement/ConsoleApplication1.cpp [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+				(Event.key.code == Keyboard::Escape))
+			{
+				window.close();
+			}
+		}
+
+
+		// prepare frame
+		window.clear();
+
```
Comment (gdxn96) [inline Spikes/movement/ConsoleApplication1.cpp]: should probably be consistent by being called update no? Reply to this comment if there is actually a difference between update/move you think needs disputing

File: Spikes/movement/Pointer.cpp [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		m_angle -= m_speed;
+	}
+	else if (Keyboard::isKeyPressed(Keyboard::Right))
+	{
+		m_angle += m_speed;
+	}
+
+	resetAngle();
+	updatePos();
+}
```
Comment (gdxn96) [inline Spikes/movement/Pointer.cpp]: again does reset angles need to be called every update? If yes, just reply to clear it up

File: Spikes/movement/Character.h [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+
+
+class Character
+{
+private:
+	//Character
+	bool m_move;
+	Vector2f m_centre;
+	Vector2f m_pos;
```
Comment (gdxn96) [inline Spikes/movement/Character.h]: is m_centre the screen centre?

File: Spikes/movement/Character.h [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+public:
+	Character();
+	Character(float, float, Color, Color, Vector2f);
+	void update();
+	void draw(RenderWindow&);
+	void resetAngles();
+	float findAngleBetween();
+
+	// Get methods
```
Comment (gdxn96) [inline Spikes/movement/Character.h]: should leave short descriptions of methods here in header above definition if not apparent, i.e. resetAngles is obvious now, but later in development it wont be, so a quick comment, update and draw are irrelevant. 

Also findAngleBetween() is an inbuilt method of Aarons Vector2D class, if we can use code thats already made do, i'm aware you probably haven't been told about the Vector2D until now so this is me notifying you. Vector2D can be found on many of the other branches and they should all be the same.

File: Spikes/movement/Character.h [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+	// Set methods
+	void setTargetAngle(float, float);
+	void setMove(bool);
```
Comment (gdxn96) [inline Spikes/movement/Character.h]: these are grand

File: Spikes/movement/ConsoleApplication1.cpp [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+	Color yellow = Color(255, 255, 51, 255);
+
+	Vector2f windowCentre = Vector2f(window.getSize().x / 2.0f,
+									 window.getSize().y / 2.0f);
+
+
+
+	//create objects
+	Pointer pointer(ptrRadius, ptrSpd, blue, windowCentre);
+	Character character(charRadius, charSpd, green, yellow, windowCentre);
```
Comment (gdxn96) [inline Spikes/movement/ConsoleApplication1.cpp]: I'd consider renaming pointer to something else, I know thats basically what it does but it does get confusing, maybe just CommandPointer/OrderPointer something like that, again use your best judgement.

File: Spikes/movement/ConsoleApplication1.cpp [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+		pointer.move();
+
+		if (Keyboard::isKeyPressed(Keyboard::Space))
+		{
+			character.setTargetAngle(pointer.getAngle(), charSpd);
+			character.setMove(true);
+		}
+
+		character.update();
+
```
Comment (gdxn96) [inline Spikes/movement/ConsoleApplication1.cpp]: updates should be done before "window.clear()" above in future, no biggy for now

File: Spikes/movement/Pointer.h [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+	CircleShape m_shape; 
+
+public:
+	Pointer(float, float, Color, Vector2f);
+	void move();
+	void draw(RenderWindow&);
+	void resetAngle();
+	void updatePos();
+	const Vector2f getPos();
+	void setSpeed(float);
```
Comment (gdxn96) [inline Spikes/movement/Pointer.h]: Try avoid using Vector2f when you can from now on, try stick with Aaron's Vector2D, a lot more builtin functions

File: Spikes/movement/Pointer.h [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+{
+private:
+	Vector2f m_centre;
+	Vector2f m_pos;
+	float m_radius;
+	float m_angle;
+	float m_speed;
+	float m_minSpeed;
+	float m_maxSpeed;
+	float m_interval;
```
Comment (gdxn96) [inline Spikes/movement/Pointer.h]: no harm commenting some of these member variables that sound a little vague i.e. m_interval, time interval, angle interval? I don't want to read through functions to understand why a variable exists.

File: Spikes/movement/Pointer.h [hunk 13]
```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#ifndef POINTER_H
+#define POINTER_H
```
Comment (gdxn96) [inline Spikes/movement/Pointer.h]: include guards + pragma once, remove one of them, across multiple files

Comment (gdxn96) [inline Spikes/movement/Pointer.h]: Only use "pragma once" actually, standards and all

File: Spikes/movement/Pointer.cpp [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+#include "Pointer.h"
+
+//using namespace sf;
+
+
+
+
+//GLOBAL VARIABLES (FOR TESTING PURPOSES)
+const int PATH_RADIUS = 200;
+const float PI = std::acos(-1);
```
Comment (gdxn96) [inline Spikes/movement/Pointer.cpp]: this is not much of a calculation to do locally in methods, I understand your logic is creating game constants, but from now on just make them locally in methods that need them, I only count 2 methods every update. Later when a game constants header file is created we can redo it, for the purposes of spikes, no need

File: Spikes/movement/ConsoleApplication1.cpp [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+
+		// prepare frame
+		window.clear();
+
+
+
+		//Update stuff
+		//-------------
+		pointer.move();
```
Comment (gdxn96) [inline Spikes/movement/ConsoleApplication1.cpp]: should probably be consistent by being called update no? Reply to this comment if there is actually a difference between update/move you think needs disputing

Comment (gdxn96) [inline Spikes/movement/ConsoleApplication1.cpp]: referring to pointer.move()

File: Spikes/movement/ConsoleApplication1.cpp [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+		}
+
+		character.update();
+
+
+
+		//Draw Stuff
+		//-----------
+		pointer.draw(window);
+		character.draw(window);
```
Comment (gdxn96) [inline Spikes/movement/ConsoleApplication1.cpp]: pass window by pointer i.e. draw(&window)

File: Spikes/movement/Character.h (hunk 3 above)
Comment (ShaneCallanan) [inline Spikes/movement/Character.h]: Note: decision made to leave targetShape in Character class

Note#2: Variables separated appropriately. Should be sufficient

File: Spikes/movement/Character.h (hunk 7 above)
Comment (ShaneCallanan) [inline Spikes/movement/Character.h]: Note: findAngleBetween() is not a method of current Vector2D class so will be left alone until updated

File: Spikes/movement/Pointer.cpp (hunk 14 above)
Comment (ShaneCallanan) [inline Spikes/movement/Pointer.cpp]: PATH_RADIUS left in.
PI changed

File: Spikes/movement/ConsoleApplication1.cpp (hunk 15 above)
Comment (ShaneCallanan) [inline Spikes/movement/ConsoleApplication1.cpp]: Leaving room for other possible functionality later.
Changed to update(), but added move() in update method

File: Spikes/Character.cpp [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+
+//GLOBAL VARIABLE (FOR TESTING PURPOSES)
+const int PATH_RADIUS = 200;
+
+
+
+Character::Character(float radius, float speed, Color green, Color yellow, 
+					 Vector2D centre) 
+:	M_PI(std::acos(-1)),
+	DEGREES(M_PI / 180.0f),
```
Comment (Lanc3) [inline Spikes/Character.cpp:14]: can declare M_PI and DEGREES like this at the top of the class
#define M_PI 3.141592654

**Pull request merged**